torchgeo/datasets/cyclone.py

```diff
@@ -96,17 +96,19 @@ def __getitem__(self, index: int) -> dict[str, Any]:
         Returns:
             data, labels, field ids, and metadata at that index
         """
-        sample = {
+        features = {
             'relative_time': torch.tensor(self.features.iat[index, 2]),
             'ocean': torch.tensor(self.features.iat[index, 3]),
             'label': torch.tensor(self.labels.iat[index, 1]),
         }
 
         image_id = self.labels.iat[index, 0]
-        sample['image'] = self._load_image(image_id)
+        sample = {'image': self._load_image(image_id)}
+        sample['label'] = features['label']
 
         if self.transforms is not None:
             sample = self.transforms(sample)
+            sample.update({x: features[x] for x in features if x != 'label'})
```

Comment:
Reason for this mentioned here- https://github.com/microsoft/torchgeo/pull/2147#discussion_r1685425105

These type of fixes won't be required once kornia/kornia#2971 makes it to a release. It allows non-kornia compatible keys to be passed.

Reply:
Hmm, this kind of seems like a hack. It's an implementation detail that `TropicalCycloneDataModule` doesn't use `TropicalCyclone(..., transform=...)`, not by design. I definitely don't think we should return different things depending on whether `transforms` is used or not.

I have a better idea. In `BaseDataModule` in `transfer_batch_to_device`, drop all keys that Kornia doesn't recognize. This will allow us to drop most changes to datasets and the `transfer_batch_to_device` in `GeoDataModule`. You can loop over all keys and drop any that don't correspond to a valid Kornia key (there should be a way to get a list of valid keys from Kornia).